addons/mail/static/src/core/web/chatter.js

```diff
@@ -269,23 +254,50 @@ export class Chatter extends Component {
     }
 
     /**
-     * @param {number} threadId
+     * Fetch data for the thread according to the request list.
      * @param {['activities'|'followers'|'attachments'|'messages'|'suggestedRecipients']} requestList
+     * @param {import("models").Thread} thread
      */
     load(
-        threadId = this.props.threadId,
-        requestList = ["followers", "attachments", "messages", "suggestedRecipients"]
+        requestList = ["followers", "attachments", "messages", "suggestedRecipients"],
+        thread = this.state.thread
```

Comment:
I would not make the param default, keep it as first param and always specify it.
And then check if the thread is different than the current one, do nothing.